File: solutions/0234-palindrome-linked-list/solution.py

class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        list = [];
        while head :
            list.append(head.val)
            head =head.next
        left = 0
        right = len(list) -1
        while left<right and list[left] == list[right] :
            left = left +1
            right =right - 1 
        return left >= right

File: solutions/0234-palindrome-linked-list/solution.js
var isPalindrome = function(head) {
   const lastVals= [];
   while(head){
    lastVals.push(head.val)
    head=head.next
   }
    let left=0;right=lastVals.length -1;
    while(left < right && lastVals[left] === lastVals[right]){
        left++;
        right--;
    }
    return left >=right
};
